From acd14ee90376fa7a698a3b58a97990f652f30a66 Mon Sep 17 00:00:00 2001
From: Billy Laws <blaws05@gmail.com>
Date: Sun, 28 Jan 2024 23:45:12 +0000
Subject: [PATCH 45/55] [compiler-rt] Mangle ARM64EC assembly function symbol
 names

ARM64EC requires that native function symbols be prefixed with a '#'
when defined in assembly, however __USER_LABEL_PREFIX__ isn't defined as
that is generally used to prefix both data and function symbols.
---
 compiler-rt/lib/builtins/assembly.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/compiler-rt/lib/builtins/assembly.h b/compiler-rt/lib/builtins/assembly.h
index 8c42fc773483..23a7c13271f0 100644
--- a/compiler-rt/lib/builtins/assembly.h
+++ b/compiler-rt/lib/builtins/assembly.h
@@ -207,7 +207,13 @@
 #define GLUE4_(a, b, c, d) a##b##c##d
 #define GLUE4(a, b, c, d) GLUE4_(a, b, c, d)
 
+#ifdef __arm64ec__
+#define QUOTE(a) #a
+#define STR(a) QUOTE(a)
+#define SYMBOL_NAME(name) STR(GLUE(#, name))
+#else
 #define SYMBOL_NAME(name) GLUE(__USER_LABEL_PREFIX__, name)
+#endif
 
 #ifdef VISIBILITY_HIDDEN
 #define DECLARE_SYMBOL_VISIBILITY(name)                                        \
-- 
2.25.1

